ludwig/trainers/trainer_lightgbm.py

```diff
@@ -804,14 +805,32 @@ def _construct_lgb_datasets(
 
         # create dataset for lightgbm
         # keep raw data for continued training https://github.com/microsoft/LightGBM/issues/4965#issuecomment-1019344293
-        lgb_train = lgb.Dataset(X_train, label=y_train, free_raw_data=False).construct()
+        try:
+            lgb_train = lgb.Dataset(X_train, label=y_train, free_raw_data=False).construct()
+        except lgb.basic.LightGBMError as e:
+            if re.search(r"special JSON characters", str(e)):
+                raise ValueError(
+                    "Some column names in the training set contain invalid characters. Please ensure column names only "
+                    "contain alphanumeric characters and underscores, then try training again."
+                )
```

Comment:
You can do `raise ValueError(...) from e` to retain the original exception as well.

Reply:
This should be addressed in [b4a0fc5](https://github.com/ludwig-ai/ludwig/pull/2879/commits/b4a0fc5e6003c633ea48a3e16517c560e8f67691)!